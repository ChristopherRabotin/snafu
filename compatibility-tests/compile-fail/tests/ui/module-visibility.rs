mod inside {
    use snafu::prelude::*;

    #[derive(Debug, Snafu)]
    #[snafu(module)]
    enum Error {
        Variant,
    }

    fn can_access_in_same_module() {
        let _ = error::VariantSnafu;
    }
}

fn cant_access_outside_of_module() {
    let _ = inside::error::VariantSnafu;
}

fn main() {}
