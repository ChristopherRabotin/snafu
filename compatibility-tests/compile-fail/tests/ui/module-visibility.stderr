error[E0603]: module `error` is private
  --> $DIR/module-visibility.rs:16:21
   |
16 |     let _ = inside::error::VariantSnafu;
   |                     ^^^^^ private module
   |
note: the module `error` is defined here
  --> $DIR/module-visibility.rs:4:21
   |
4  |     #[derive(Debug, Snafu)]
   |                     ^^^^^
   = note: this error originates in the derive macro `Snafu` (in Nightly builds, run with -Z macro-backtrace for more info)
